src/fem/dofhandler.hh

```diff
@@ -284,47 +202,42 @@ public:
 
     /**
      * @brief Distributes degrees of freedom on the mesh needed
-     * for the given finite elements.
+     * for the given discrete space.
      *
+     * By default, the dof handler is parallel, meaning that each
+     * processor has access to dofs on the local elements and on one
+     * layer of ghost elements (owned by neighbouring elements).
+     * This can be changed by setting @p sequential to true.
+     * 
      * The additional parameter @p offset allows to reserve space
      * for another finite element dofs in the beginning of the
      * global dof vector.
      *
-     * @param fe1d The 1D finite element.
-     * @param fe2d The 2D finite element.
-     * @param fe3d The 3D finite element.
-     * @param offset The offset.
+     * @param ds         The discrete space consisting of finite elements for each mesh element.
+     * @param sequential If true then each processor will have information about all dofs.
+     * @param offset     The offset.
      */
-    void distribute_dofs(FiniteElement<1> &fe1d,
-    		FiniteElement<2> &fe2d,
-    		FiniteElement<3> &fe3d,
-    		const unsigned int offset = 0);
+    void distribute_dofs(std::shared_ptr<DiscreteSpace> ds,
+                         bool sequential = false,
+                         const unsigned int offset = 0);
```

Comment:
Do we need offset anymore?

Reply:
Probably not. This was originally meant for systems of finite elements, which are now possible through FESystem.